python/dgllife/utils/complex_to_graph.py

```diff
@@ -56,6 +58,148 @@ def get_atomic_numbers(mol, indices):
+        Default to None.
+    distance_bins : list of float
+        Sequence of distance edges to determine the edge types.
+        The first edge type is connected between atoms of distance less than `distance_bins[0]`.
+        The length matches the number of edge types to be constructed.
+        Default `[1.5, 2.5, 3.5, 4.5]`.
+    max_num_neighbors : int
+        Maximum number of neighbors allowed for each atom when constucting KNN graph. Default to 4.
+    strip_hydrogens : bool
+        Whether to exclude hydrogen atoms. Default to False.
+
+    Returns
+    -------
+    complex_bigraph : DGLGraph
+        Bigraph with ligand and protein (pocket) combined and canonical features extracted.
+        The atom features are stored as DGLGraph.ndata['h'].
+        The edge types are stored as DGLGraph.edata['e'].
+        The bigraphs of ligand and protein are batched together as one complex graph.
+    complex_knn_graph : DGLGraph
+        K-nearest-neighbor graph with ligand and protein (pocket) combined and edge features extracted based on distances.
+        The edge types are stored as DGLGraph.edata['e'].
+        The knn-graphs of ligand and protein are batched together as one complex graph.
+
+    """
+
+    assert ligand_coordinates is not None, 'Expect ligand_coordinates to be provided.'
+    assert protein_coordinates is not None, 'Expect protein_coordinates to be provided.'
+    if max_num_ligand_atoms is not None:
+        assert max_num_ligand_atoms >= ligand_mol.GetNumAtoms(), \
+            'Expect max_num_ligand_atoms to be no smaller than ligand_mol.GetNumAtoms(), ' \
+            'got {:d} and {:d}'.format(max_num_ligand_atoms, ligand_mol.GetNumAtoms())
+    if max_num_protein_atoms is not None:
+        assert max_num_protein_atoms >= protein_mol.GetNumAtoms(), \
+            'Expect max_num_protein_atoms to be no smaller than protein_mol.GetNumAtoms(), ' \
+            'got {:d} and {:d}'.format(max_num_protein_atoms, protein_mol.GetNumAtoms())
+
+    if strip_hydrogens:
+        # Remove hydrogen atoms and their corresponding coordinates
+        ligand_atom_indices_left = filter_out_hydrogens(ligand_mol)
+        protein_atom_indices_left = filter_out_hydrogens(protein_mol)
+        ligand_coordinates = ligand_coordinates.take(ligand_atom_indices_left, axis=0)
+        protein_coordinates = protein_coordinates.take(protein_atom_indices_left, axis=0)
+    else:
+        ligand_atom_indices_left = list(range(ligand_mol.GetNumAtoms()))
+        protein_atom_indices_left = list(range(protein_mol.GetNumAtoms()))
+
+    # Compute number of nodes for each type
+    if max_num_ligand_atoms is None:
+        num_ligand_atoms = len(ligand_atom_indices_left)
+    else:
+        num_ligand_atoms = max_num_ligand_atoms
+
+    if max_num_protein_atoms is None:
+        num_protein_atoms = len(protein_atom_indices_left)
+    else:
+        num_protein_atoms = max_num_protein_atoms
```

Comment:
`num_ligand_atoms` and `num_protein_atoms` are not used anywhere. Same for `max_num_ligand_atoms` and `max_num_protein_atoms`.